### PR TITLE
Add user events for supporter and username changes

### LINKED ISSUE #3173: Supporter tags and username changes aren't displaying recent events in profile
As discussed.

### PR DESCRIPTION
**Pull request by notbakaneko**
fixes #3173

### CODE REVIEW

File: app/Models/Event.php [hunk 1]
```diff
@@ -313,4 +364,13 @@ public function scopeRecent($query)
     {
         return $query->orderBy('date', 'desc')->limit(5);
     }
+
+    private static function userParams($user)
+    {
+        return [
+            'id' => $user->getKey(),
+            'username' => $user->username,
```
Comment (nanaya) [inline app/Models/Event.php]: escape?

File: app/Models/Event.php [hunk 2]
```diff
@@ -70,6 +70,57 @@ public static function generate($type, $options)
...
                     'epicfactor' => 8,
                 ];
+
+                break;
+
+            case 'usernameChange':
+                $user = static::userParams($options['user']);
+                $history = $options['history'];
+                $params = [
+                    'text' => "<b><a href='{$user['url']}'>{$history->username_last}</a></b> has changed their username to {$history->username}!",
```
Comment (nanaya) [inline app/Models/Event.php]: Stuff from `$history` still needs escaping.

File: app/Models/Event.php [hunk 3]
```diff
@@ -70,6 +70,58 @@ public static function generate($type, $options)
...
+                break;
+
+            case 'usernameChange':
+                $user = static::userParams($options['user']);
+                $oldUsername = e($options['history']->username_last);
+                $newUsername = e($options['history']->username);
+                $params = [
+                    'text' => "<b><a href='{$user['url']}'>{$oldUsername}</a></b> has changed their username to {$newUsername}!",
+                    'user_id' => $user['id'],
+                    'date' => $history->timestamp,
```
Comment (nanaya) [inline app/Models/Event.php]: explodesssss

Comment (notbakaneko) [reply app/Models/Event.php]: oh whoops 💃

**Pull request merged**